tools/sql-extensions/postgres/driver.go

```diff
@@ -83,11 +81,8 @@ func (d *driver) GetDriverName() string {
 	return driverName
 }
 
-func (d *driver) CreateDBConnection(driverName, host string, port int, user string, passwd string, database string) (*sqlx.DB, error) {
-	if database == "" {
-		database = "postgres"
-	}
-	db, err := sqlx.Connect(driverName, fmt.Sprintf(dataSourceNamePostgres, user, passwd, host, port, database))
+func (d *driver) CreateDBConnection(cfg *config.SQL) (*sqlx.DB, error) {
+	db, err := storage.NewSQLDB(cfg)
```

Comment:
This will not work unless you do it after my PR: https://github.com/uber/cadence/pull/2836 
Also need to keep the logic to use "postgres" when database is empty to act as admin for creating/dropping database for postgres.